### PR TITLE
🔥 Add support for git submodules

### PR DESCRIPTION
**Pull request by akhilerm**
generate link to submodule repository if the selected file is from a git submodule

Signed-off-by: Akhil Mohan <akhilerm@gmail.com>

### CODE REVIEW

**Comment by akhilerm** [comment]
I have kept the release at 1.0.5 itself since I didnt see a release generated for the last commit, i.e support for spaces

**Comment by akhilerm** [comment]
@lunakoly Can this change be reviewed?

**Comment by lunakoly** [comment]
Sure, I wrote 2 comments

**Comment by akhilerm** [comment]
@lunakoly  Are the comments not submitted? I cant see them on the changes

**Review by lunakoly — CHANGES_REQUESTED** [review]

**Comment by lunakoly** [comment]
> @lunakoly Are the comments not submitted? I cant see them on the changes

Oh, I'm sorry, I didn't notice that. And [it seems I'm not the only one](https://github.com/community/community/discussions/10369)...

File: src/main/kotlin/com/github/lunakoly/quicklink/repository/RepositoryInfoBuilders.kt [hunk 1]
```diff
@@ -44,12 +44,17 @@ fun getRepositoryInfoAsGit(repo: GitRepository): RepositoryInfo {
     return RepositoryInfo(repo.root, branch, commitHash, map)
 }
 
-fun getRepositoryInfo(project: Project): RepositoryInfo {
+fun getRepositoryInfo(project: Project): ArrayList<RepositoryInfo> {
+    // a project can have multiple repositories reported, if there are
+    // submodules in it. info about each of those repositories will be
+    // fetched adn returned
+    val repositories = ArrayList<RepositoryInfo>()
```
[the remark this answers is not in the document]
Comment (akhilerm) [reply src/main/kotlin/com/github/lunakoly/quicklink/repository/RepositoryInfoBuilders.kt]: Done.

File: src/main/kotlin/com/github/lunakoly/quicklink/actions/CopyLineLinkAction.kt [hunk 2]
```diff
@@ -51,7 +51,19 @@ class CopyLineLinkAction : AnAction() {
...
+
+        // sort in descending order by the root path so that the first match will be
+        // the nearest parent of the given file
+        repositoryInfoList.sortByDescending {
+            it.root.path.length
+        }
+
+        // returns the repository in which the file is a part of. It can
+        // either be the submodule or the original repository
+        val repositoryInfo = repositoryInfoList.find {
```
[the remark this answers is not in the document]
Comment (akhilerm) [reply src/main/kotlin/com/github/lunakoly/quicklink/actions/CopyLineLinkAction.kt]: Done

**Pull request merged**